src/components/Vote.vue

```diff
@@ -28,25 +26,29 @@ export default {
   components: {
     Results,
   },
-  firebase: {
-    polls: {
-      source: polls,
-      readyCallback() {
-        this.isLoading = false;
-        this.poll = this.polls.filter(poll => poll['.key'] === this.$route.params.id)[0];
+  firebase() {
+    return {
+      poll: {
+        source: db.ref().child(`/${this.$route.params.id}`),
+        asObject: true,
+        readyCallback() {
+          this.isLoading = false;
+        },
       },
-    },
+    };
   },
   methods: {
     vote(poll, choice) {
+      const dbRef = db.ref().child(`/${this.$route.params.id}`);
```

Comment:
If I understand the docs correctly it should be possibly to simplify this logic even further by using
```
.child(`/${this.$route.params.id}/${choice}/votes`)
```